web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js

```diff
@@ -275,17 +288,27 @@ patch(PivotModel.prototype, {
         const fieldNames = Object.keys(this.metaData.fields);
         for (const fieldName of fieldNames) {
             const field = this.metaData.fields[fieldName];
-            if (field.__computed_id) {
-                const cm = this._computed_measures.find((item) => {
-                    return item.id === field.__computed_id;
-                });
-                if (!cm) {
-                    delete this.metaData.fields[fieldName];
-                    delete this.metaData.measures[fieldName];
-                    this.metaData.activeMeasures = this.metaData.activeMeasures.filter(
-                        (item) => item !== fieldName
+            if (field) {
+                if (field.__computed_id) {
+                    const cm = this._computed_measures.find(
+                        (item) => item.id === field.__computed_id
+                    );
+                    if (!cm) {
+                        delete this.metaData.fields[fieldName];
+                        delete this.metaData.measures[fieldName];
+                        this.metaData.activeMeasures =
+                            this.metaData.activeMeasures.filter(
+                                (item) => item !== fieldName
+                            );
+                    }
+                } else {
+                    console.warn(
+                        `Field "${fieldName}" is missing __computed_id`,
+                        field
                     );
                 }
+            } else {
+                console.error(`Field "${fieldName}" is undefined in metaData.fields`);
```

Comment:
Are these warnings and errors of any use? If this is a corner case where the code is executed but those variables aren't ready, probably the best thing to do is just to ignore it...

Reply:
I think the test is not aligned with the code, there maybe some promises not functioning.

I tried but I failed as I am not the maintainer, I need help to improve the test case.

Reply:
@CarlosRoca13 is on holidays these days. Maybe when he comes back could give you some hints :wink:

Reply:
@chienandalu  yes, thanks to you too much for being interested.
Happy New Year